flatbuffers/flatbuffers_case_study.md

```diff
@@ -0,0 +1,151 @@
+Case study of Javascript Flat Buffers
+================================
+
+Data serialization is very common process in web application. So common that no one thinks of it. You just pull data from the DB, and your server and client frameworks just handle this thing for you. The server serializes the data into JSON, and the client parses it to JS object. This flow is simple and natural but, as everything, when taken to the extreme start to show scaling issues.
+
+Whats an 'extreme' for data serialization? A lot of data. Serializing and deseriazling tens of thousands of objects takes it toll from the server, and more importantly, from the single threaded browser page.
+
+A possible solution for this scaling issue are FlatBuffers. We've utilized this tool to do our heavy lifting and lived to tell about it.
+
+# Who are we?
+Before diving into FlatBuffers internals let me tell you a bit about us. Why do we even need to transfer so many objects to the browser?
+
+We are [Jaco](https://getjaco.com), a UX analytics tool that enable you to watch your user sessions exactly as they happen - a movie clip of a users screen while he or she browse your site.
+
+The basic building blocks of a user session recording tools are:
+* A recorder
+* Processing pipeline
+* Recordings storage
+* Recordings player
+
+The recorder have the ability to capture any event or DOM mutation in a page, and stream the recorded entities to our servers. As you can imagine a session might consist of a huge number of events. Every mouse move, every scroll, every change to the page itself must be recorded. This can easily mount to ten or hundreds of thousands of data points.
+
+All this data must be stored and accessible for the recorders counterpart - the player. The player knows how to take the mass of events and reconstruct them into a watchable clip.
+
+This undertaking raises many challenges, one of which is to transfer all of those events from the recorded user to our servers, and from our servers to the player. The efficiency of this process is measured both by the size of the transfered data and by the time it takes to parse it, measures that JSON does not excel in.
```

Comment:
I would also add a background check we did of FB vs other similar frameworks we did and why we chose flatbuffers..